lib/Parse/ParsePattern.cpp

```diff
@@ -888,6 +879,56 @@ void Parser::parseAsyncThrows(
   }
 }
 
+/// Parse a type in parentheses after a `throws` or `rethrows`
+///
+ParserResult<TypeRepr> Parser::parseThrowsType() {
+  // look for the left paren for throws (_type_)
+  BacktrackingScope backtrack(*this);
+  
+  ParserResult<TypeRepr> throwsTypeRepr;
+  
+  if (Tok.is(tok::l_paren)) {
+    consumeToken();
+    throwsTypeRepr = parseTypeIdentifier();
+    if (!throwsTypeRepr.isParseError()) {
```

Comment:
In case, the type parsing had an error, we probably also need a diagnostic, but maybe `parseType...()` already emits one.

Reply:
yes, as above I need to add some error cases and diagnostics for things. `parseTypeIdentifier` will generate diagnostics if it finds unparsable stuff there. I also need to add that to the tests.